cmd/snap-confine/context-support-test.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "context-support.h"
+#include "context-support.c"
+
+#include "../libsnap-confine-private/test-utils.h"
+
+#include <glib.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+// Set alternate context directory
+static void set_context_dir(const char *dir)
+{
+	sc_context_dir = dir;
+}
+
+static void set_fake_context_dir()
+{
+	char *ctx_dir = NULL;
+	ctx_dir = g_dir_make_tmp(NULL, NULL);
+	g_assert_nonnull(ctx_dir);
+	g_test_queue_free(ctx_dir);
+
+	g_test_queue_destroy((GDestroyNotify) rm_rf_tmp, ctx_dir);
+	g_test_queue_destroy((GDestroyNotify) set_context_dir, SC_CONTEXT_DIR);
+
+	set_context_dir(ctx_dir);
+}
+
+static void create_dumy_context_file(const char *snap_name,
+				     const char *dummy_context)
+{
+	char path[256];
+	FILE *f;
+	int n;
+
+	snprintf(path, sizeof(path), "%s/snap.%s", sc_context_dir, snap_name);
+
+	f = fopen(path, "w");
+	g_assert_nonnull(f);
+
+	n = fwrite(dummy_context, 1, strlen(dummy_context), f);
+	g_assert_cmpint(n, ==, strlen(dummy_context));
+
+	fclose(f);
+}
+
+static void test_maybe_set_context_environment__null()
+{
+	setenv("SNAP_CONTEXT", "bar", 1);
```

Comment:
This clobbers `SNAP_CONTEXT` during testing. This might affect other places. Can you please move this to a sub-process or add cleanup code?